packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -430,7 +430,15 @@ export class CodeModelBuilder {
     const deduplicateName = (schema: Schema) => {
       const name = schema.language.default.name;
       // skip models under "com.azure.core."
-      if (name && !schema.language.java?.namespace?.startsWith("com.azure.core.")) {
+      if (
+        name &&
+        // skip models under "com.azure.core."/"Azure.", if branded
+        !(
+          this.isBranded() &&
+          (schema.language.java?.namespace?.startsWith("com.azure.core.") ||
+            schema.language.default?.namespace?.startsWith("Azure."))
```

Comment:
limit to Azure.Core as well?

Reply:
I've got mind on TypeSpec model under `Azure.ResourceManager.` as well. Avoid touch them as well.